pkg/types/baremetal/cache.go

```diff
@@ -0,0 +1,46 @@
+package baremetal
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+)
+
+// BmhCacheListerWatcher is an object that wraps the listing and wrapping
+// functionality for baremetal host resources.
+type BmhCacheListerWatcher struct {
+	Resource dynamic.ResourceInterface
+}
+
+// List returns a list of baremetal hosts as dynamic objects.
+func (bc BmhCacheListerWatcher) List(options metav1.ListOptions) (runtime.Object, error) {
+	list, err := bc.Resource.List(context.TODO(), options)
+	if err != nil {
+		if err.Error() == "the server could not find the requested resource" {
+			return &unstructured.UnstructuredList{}, nil
+		}
+	}
+	return list, err
+}
+
+// Watch starts a watch over baremetal hosts.
+func (bc BmhCacheListerWatcher) Watch(options metav1.ListOptions) (watch.Interface, error) {
+	w, err := bc.Resource.Watch(context.TODO(), options)
+	if err != nil {
+		if err.Error() == "the server could not find the requested resource" {
```

Comment:
I'd prefer to try to convert this to a `*k8s.io/apimachinery/pkg/api/errors.StatusError`, then we can check for `statusErr.Status().Reason == metav1.StatusReasonNotFound`.

Reply:
This guard protects against the case where the `BareMetalHost` resource definition isn't in place yet.

Reply:
Right, my objection is to comparing the text of an error message (specifically, it's [this one](https://github.com/openshift/installer/blob/master/vendor/k8s.io/apimachinery/pkg/api/errors/errors.go#L448)) rather than doing something more structured.

Reply:
I see!  I think I misread your comment as `metal3.StatusReasonNotFound` and didn't think we could check for that if we didn't have any metal3 CRDs.

Reply:
Done